### DISCUSSION TITLE
Force extension to be installed from `https`

### DISCUSSION THREAD

**Discussion by robertdlangford** [duckdb/duckdb#12695] [Q&A]
In trying to load extensions (including: `httpfs` & `spatial`) on a Windows server I'm getting the following 

```
D install httpfs;
IO Error: Failed to download extension "httpfs" at URL "http://extensions.duckdb.org/v1.0.0/windows_amd64/httpfs.duckdb_extension.gz"
Extension "httpfs" is an existing extension.
 (ERROR Read)
```

I think it's something to do with the `http` (rather than being `https`) as `http` links are blocked our corporate admin.  Is there any way to force DuckDB to use the `https` version?

I know I can manually download the extension, but I'm trying to package this up as a portable process so would like to avoid that step if possible.

Thanks.

---

Using Python and Command line
v1.0.0 1f98600c2c

**Comment by szarnyasg** [comment]
Hi @robertdlangford, this is a bit of a catch-22 problem. At the moment, your best bet is to download the `httpfs` package via other means, e.g. via Python or wget/curl.

**Reply by Tishj** [reply]
I think as a work around, I believe that it's possible to provide a URL to download from, in the `install` statement
`INSTALL 'https://extensions.duckdb.org/v1.0.0/windows_amd64/httpfs.duckdb_extension.gz';`

That might work?
Though it's possible that only works for local paths, I have not verified

Perhaps this `INSTALL httpfs FROM '<path without httpfs.duckdb_extension.gz>'`  could be another thing to try

**Reply by robertdlangford** [reply]
Thanks @szarnyasg for your response.  Seems strange to not be using HTTPS.  However, I've been able to get the URL added to a whitelist so is working fine now.

Unfortunately @Tishj that means I'm not sure I can test those solutions as it all works now.  Thanks for the suggestions anyway, maybe they will help someone else in the future.

🙏
